iphone/Maps/Classes/ClipboardMonitor.mm

```diff
@@ -0,0 +1,107 @@
+#import "MapsAppDelegate.h"
+#import "ClipboardMonitor.h"
+#import "UIKitCategories.h"
+#import "MapsAppDelegate.h"
+#import <CoreApi/DeepLinkParser.h>
+#include <CoreApi/Framework.h>
+#import <CoreApi/DeepLinkData.h>
+#import "UIKitCategories.h"
+#import <CoreApi/DeepLinkSearchData.h>
+
+@interface ClipboardWatcher ()
+
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, copy) NSString *previousClipboardText;
+@property (nonatomic, assign) BOOL isErrorPromptShown;
+
+@end
+
+@implementation ClipboardWatcher
+
+- (void)startWatching:(UIWindow *)window {
+    NSLog(@"clip watching started");
+
+    UIPasteboard *pasteboard = [UIPasteboard generalPasteboard];
+    NSString *currentClipboardText = pasteboard.string;
+    NSLog(@"clip clipboard checking for changes %@  and  %@",currentClipboardText, self.previousClipboardText);
+
+    [pasteboard setString:@""];
+    [self checkClipboardWithText:currentClipboardText window:window];
+}
+
+- (void)checkClipboardWithText:(NSString *)copiedText window:(UIWindow *)window {
+    NSLog(@"clip API CALL PROCEEDING %@", copiedText);
+
+    // Check if the copied text is a URL
+    if ([self isURL:copiedText]) {
```

Comment:
What kind of urls are supported now by a proxy? Google Maps? Something else? Only these URLs should be explicitly detected in the pasteboard and passed to the proxy.

Reply:
Right now Google Maps URIs are supported by the proxy, I have updated the code such that it would make a call only if the host name corresponds to google maps' hostnames.